core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java

```diff
@@ -328,26 +329,173 @@ default MediaType negotiatedProduceType() {
     }
 
     /**
-     * Returns the amount of time allowed until receiving the current {@link Request} and sending
-     * the corresponding {@link Response} completely.
+     * Returns the amount of time allowed from the start time of the {@link Request} until receiving
+     * the current {@link Request} and sending the corresponding {@link Response} completely.
      * This value is initially set from {@link ServiceConfig#requestTimeoutMillis()}.
      */
     long requestTimeoutMillis();
 
     /**
-     * Sets the amount of time allowed until receiving the current {@link Request} and sending
-     * the corresponding {@link Response} completely.
+     * Clears the request timeout.
+     */
+    void clearRequestTimeout();
+
+    /**
+     * Sets the amount of time allowed from the start time of the request until receiving
+     * the current {@link Request} and sending the corresponding {@link Response} completely.
```

Comment:
Please make the same documentation changes as what I suggested in `ClientRequestContext`.